src/cubical/Cof.ml

```diff
@@ -40,11 +41,19 @@ let meet2 phi psi =
 let join l = List.fold_left join2 bot l
 let meet l = List.fold_left meet2 top l
 
+let neg_eq ~dim0 ~dim1 r1 r2 =
+  join2
+    (meet [eq r1 dim0; eq r2 dim1])
+    (meet [eq r1 dim1; eq r2 dim0])
+
+let neg phi = Cof (Neg phi)
+
 let rec reduce =
   function
-  | Cof (Join phis) -> join @@ List.map reduce phis
-  | Cof (Meet phis) -> meet @@ List.map reduce phis
+  | Cof (Join phis) -> join @@ List.map (reduce) phis
+  | Cof (Meet phis) -> meet @@ List.map (reduce) phis
```

Comment:
what do these parentheses do?

Reply:
LOL it's some leftovers from earlier experiments. BTW the `reduce` function itself seems to be useless.